src/Raven.Server/Documents/Handlers/Processors/Smuggler/AbstractSmugglerHandlerProcessorForExport.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System;
+using System.Threading.Tasks;
+using JetBrains.Annotations;
+using Raven.Client.Documents.Smuggler;
+using Sparrow.Json;
+
+namespace Raven.Server.Documents.Handlers.Processors.Smuggler
+{
+    internal abstract class AbstractSmugglerHandlerProcessorForExport<TRequestHandler, TOperationContext> : AbstractDatabaseHandlerProcessor<TRequestHandler, TOperationContext>
+        where TRequestHandler : AbstractDatabaseRequestHandler<TOperationContext>
+        where TOperationContext : JsonOperationContext
+    {
+        public AbstractSmugglerHandlerProcessorForExport([NotNull] TRequestHandler requestHandler) : base(requestHandler)
+        {
+        }
+
+        protected abstract ValueTask ExportAsync(JsonOperationContext context, long? operationId);
+
+        public override async ValueTask ExecuteAsync()
+        {
+            var result = new SmugglerResult();
+            
+            using (ContextPool.AllocateOperationContext(out JsonOperationContext context))
+            {
+                var operationId = RequestHandler.GetLongQueryString("operationId", false);
+                try
+                {
+                    await ExportAsync(context, operationId);
+                }
+                catch (Exception e)
+                {
+                    if (Logger.IsOperationsEnabled)
+                        Logger.Operations("Export failed .", e);
+
+                    result.AddError($"Error occurred during export. Exception: {e.Message}");
+                    await RequestHandler.WriteResultAsync(context, result, RequestHandler.ResponseBodyStream());
```

Comment:
remove WriteResultAsync from RequestHandler and move it here?

Reply:
moved